tap/tapify.py

```diff
@@ -355,3 +356,54 @@ def tapify(
 
     # Initialize the class or run the function with the parsed arguments
     return class_or_function(*class_or_function_args, **class_or_function_kwargs)
+
+
+def tapify_with_subparsers(class_: Type):
+    # Create a Tap class with subparsers defined by the class_'s methods
+    docstring = _docstring(class_)
+    param_to_description = {param.arg_name: param.description for param in docstring.params}
+    args_data = _tap_data(class_, param_to_description, func_kwargs={}).args_data
+
+    subparser_dest = "_tap_subparser_dest"
+
+    class TapWithSubparsers(_tap_class(args_data)):
+        def configure(self):  # TODO: understand why overriding _configure is wrong
+            self.add_subparsers(
+                help="sub-command help",  # TODO: prolly should be user-inputted instead
+                required=True,  # If not required just use tapify
+                dest=subparser_dest,  # Need to know which subparser (i.e., which method) is being hit by the CLI
+            )
+            for method_name in dir(class_):
+                method = getattr(class_, method_name)
+                if method_name.startswith("_") or not callable(method):
+                    # TODO: maybe the user can input their own function (method_name: str -> bool) for deciding whether
```

Comment:
Maybe add a decorator to mark methods as subparsers then this can all be integrated into the existing `tapify` implementation.

--JK

Reply:
Neat. That sounds like a better interface